src/Chainweb/CutDB.hs

```diff
@@ -480,6 +519,33 @@ cutHashesToBlockHeaderMap headerStore payloadStore hs = do
                     return $! Left cv
                 e -> throwM e
 
+-- | Yield a `Cut` only when it is determined that `CutDb` has sync'd with
+-- remote peers enough. "Enough" is a measure of "closeness" determined from the
+-- `ChainGraph` implied by the given `ChainwebVersion`. In essence, if our
+-- current Cut is...:
+--
+--   * below the threshold: Spin via STM (this has the affect of pausing POW mining).
+--   * above the threshold: Yield the `Cut`.
+--   * even higher than the network: We are either a "superior" fork, or we are in
+--     initial network conditions where there is no real consensus yet. In this
+--     case, we yield a `Cut`.
+--
+consensusCut :: CutDb cas -> IO Cut
+consensusCut cutdb = atomically $ do
+    nh  <- avgNetworkHeight <$> readTVar (_cutPeerHeights cutdb)
+    cur <- _cutStm cutdb
+    let !currentHeight = _cutHeight cur
+        !thresh = int . catchupThreshold $ _chainwebVersion cutdb
+        !mini = bool (nh - thresh) 0 $ thresh > nh
+    when (currentHeight < mini) retry
+    pure cur
+
+-- | The distance from the true Cut within which the current node could be
+-- considered "caught up".
+--
+catchupThreshold :: ChainwebVersion -> Natural
+catchupThreshold = (2 *) . diameter . _chainGraph
```

Comment:
It'd probably be easier to inline this here or as a data entry in the chainweb version instead of calculating this every time we  need to establish  a consensusCut. That way, we can think of chainweb versions as having a tuneable "catchup threshold" built in.

Reply:
Memoize?